src/main/java/com/spotify/docker/client/messages/HostConfig.java

```diff
@@ -627,7 +628,22 @@ public Builder hostPidMode() {
     
     public abstract Builder storageOpt(Map<String, String> tmpfs);
 
-    public abstract HostConfig build();
+    abstract HostConfig autoBuild();
+
+    public HostConfig build() {
+      final HostConfig hostConfig = autoBuild();
+      validateExtraHosts(hostConfig.extraHosts());
+      return hostConfig;
+    }
+  }
+
+  private static void validateExtraHosts(final List<String> extraHosts) {
+    if (extraHosts != null) {
+      for (final String extraHost : extraHosts) {
+        checkArgument(extraHost.contains(":"),
+            String.format("extra host arg '%s' must contain a ':'", extraHost));
```

Comment:
no need to call String.format yourself, there is an overload of checkArgument that does:

```
public static void checkArgument(boolean expression,
                 @Nullable
                 String errorMessageTemplate,
                 @Nullable
                 Object... errorMessageArgs)
```